### PR TITLE
chore: Remove method getByPrograms(List<Program>) [DHIS2-17712]

### PR DESCRIPTION
**Pull request by muilpp**
This PR continues the work of moving tracker-related code out of the API module. Specifically it:

- Removes the `getByPrograms` method from `EnrollmentStore` and its implementation. The logic to get event programs from the database is now a private method in `EnrollmentSupplier`, since we need to access all enrollments regardless of user authorization and don't need/want this method in our exporters.
- Removes the `countEnrollments` method from `EnrollmentStore` and makes it private in its implementation, as it's only used internally.
- Removes a unit test from `EnrollmentSupplierTest` because it can't be properly mocked anymore. The same behavior is tested in `EventImportValidationTest.testTrackerAndProgramEventUpdateSuccess`.

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java [hunk 1]
```diff
@@ -73,4 +70,13 @@ public void preheatAdd(TrackerObjects trackerObjects, TrackerPreheat preheat) {
           });
     }
   }
+
+  private List<Enrollment> getEnrollmentsByProgram(List<Program> programs) {
+    TypedQuery<Enrollment> query =
+        entityManager.createQuery(
+            "FROM Enrollment e WHERE e.deleted = false AND e.program IN (:programIds)",
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java:77]: Can we settle on using lowercase in our SQL 😋 it's more comfortable to type and does not look like screaming 😂

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java [hunk 2]
```diff
@@ -60,11 +59,9 @@ public void preheatAdd(TrackerObjects trackerObjects, TrackerPreheat preheat) {
     if (programsWithoutRegistration.isEmpty()) {
       programsWithoutRegistration = programStore.getByType(ProgramType.WITHOUT_REGISTRATION);
     }
+
     if (!programsWithoutRegistration.isEmpty()) {
-      List<Enrollment> enrollments =
-          DetachUtils.detach(
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java:65]: Is that not still needed? or are we not getting a proxied enrollment with the `EntityManager`?

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java [hunk 3]
```diff
@@ -73,4 +70,13 @@ public void preheatAdd(TrackerObjects trackerObjects, TrackerPreheat preheat) {
           });
     }
   }
+
+  private List<Enrollment> getEnrollmentsByProgram(List<Program> programs) {
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java:74]: This is not strictly related to this PR but could we make the query just get all the enrollments with programs without registration instead of getting the programs and passing the ids?

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java (hunk 2 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java:65]: Agreed. I think we still need to detach the enrollments.

Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java:65]: yep, nice catch!

**Review by enricocolasante — APPROVED** [review]

**Review by teleivo — APPROVED** [review]
